lib/modules/saveComments.js

```diff
@@ -54,6 +76,117 @@ module.go = async () => {
 	}
 };
 
+// Comment save on reply implementation
+
+function addReplyListener(thing) {
+	if (thing) {
+		const replyButton = thing.element.querySelector('a.access-required[data-event-action=comment]');
+		replyButton.addEventListener('click', saveCommentHandler, false);
+	}
+}
+
+function addPreviousCommentButton(thing) {
+	// Add a button to the button list to view
+	// the previous versions of the comment
+	const buttonsLists = thing.element.querySelector('ul.buttons');
+	const template = string.html`
+    <li>
+        <a href="#" class="noCtrlF">${i18n('commentSaveOnReplyButton')}</a>
+    </li>`;
+	const link = template.querySelector('a');
+	link.addEventListener('click', loadPreviousCommentHandler, false);
+	buttonsLists.appendChild(template);
+}
+
+function saveCommentHandler() {
+	const commentThing = Thing.from(this);
+	if (commentThing) {
+		saveComment(commentThing, true);
+	}
+}
+
+function hidePreviousCommentHandler(event) {
+	event.preventDefault();
+	const commentArea = Thing.from(this);
+	if (commentArea) {
+		const md = commentArea.element.getElementsByClassName('md')[0];
+		const prev = md.getElementsByClassName('res-comment-saved-text')[0];
+		md.removeChild(prev);
+		updateButton(this, true);
+	}
+}
+
+async function loadPreviousCommentHandler(event: Event) {
+	event.preventDefault();
+	const commentThing = Thing.from(this);
+	if (commentThing) {
+		const commentElement = commentThing.element;
+		const comment = await getComment(getId(commentThing));
```

Comment:
Note that `getId` doesn't work on deleted comments. I think being able to also restore those comments would be useful.

Reply:
What do you think about storing the reply comment id as a property of saved comments, then in the event of a deleted comment retrieving saved comments by their child (reply) id? Just brainstorming.

Reply:
It appears the id does remain visible in deleted comments via `<div class="reportform report-t1_drow9l2"></div>`. I will try to extend functionality to deleted comments.